.github/workflows/build.yml

```diff
@@ -64,3 +64,11 @@ jobs:
         github_token: ${{ secrets.GITHUB_TOKEN }}
         publish_dir: ./public
         force_orphan: true
+    - name: Deploy PR preview
+      if: github.event_name == 'pull_request' && github.repository == github.event.pull_request.head.repo.full_name # if this build is a PR build and the PR is NOT from a fork
+      uses: JamesIves/github-pages-deploy-action@releases/v3
```

Comment:
this is a different action from the one above, which is maybe a bit funny. Not familiar with the other one.

As a further improvement, the step deploying main should protect `previews` folder from cleanup. This is needed if there are multiple PRs open to avoid that merging one removes the previews of all the others. Arguably not a problem at our current PR volume.

In this action, this would be done by adding
```
                  CLEAN_EXCLUDE: |
                    previews/*
```

but not sure about the `peaceiris/actions-gh-pages@v3`